File: java/app/src/main/java/io/justtrack/demo/IronSourceActivity.java
package io.justtrack.demo;

import android.app.Activity;
import android.os.Bundle;

import androidx.annotation.Nullable;

import java.util.concurrent.ExecutionException;
import java.util.concurrent.Future;

import io.justtrack.JustTrackSdk;
import io.justtrack.JustTrackSdkBuilder;
import io.justtrack.SdkBuilder;
import io.justtrack.UserIdSource;

/** @noinspection ConstantValue*/
public class IronSourceActivity extends Activity {
    public final String ironSourceUserId = "..your user id";

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_ironsource);
        SdkBuilder builder = new JustTrackSdkBuilder(this, MainActivity.TOKEN)
                .setUserId("unique_user_id");
        if (ironSourceUserId != null) {
            builder.setEnableIronSourceIntegration(true, ironSourceUserId);
        } else {
            builder.setEnableIronSourceIntegration(true, UserIdSource.JustTrack);
        }

        builder.build();
    }

    /**
     * If you haven't set up the IronSource SDK using "setEnableIronSourceIntegration,"
     * you can integrate IronSource by using "integrateIronSource" after creating the SDK.
     *
     * @noinspection unused
     */
    private void integrateIronSource(JustTrackSdk sdk) throws ExecutionException, InterruptedException {
        if (ironSourceUserId == null) {
            Future<Void> integrationFuture = sdk.integrateWithIronSource(UserIdSource.JustTrack);
            // if you need to know whether the integration succeeded and is done:
            integrationFuture.get();
        } else {
            sdk.integrateWithIronSource(ironSourceUserId);
        }
        // if no exception is thrown, the integration was successful
    }
}

File: java/app/src/main/java/io/justtrack/demo/IronSourceActivity.kt
package io.justtrack.demo

import android.app.Activity
import android.os.Bundle
import io.justtrack.IronSourceUserIdSource
import io.justtrack.JustTrackSdk
import io.justtrack.JustTrackSdkBuilder
import io.justtrack.SdkBuilder

/**
 * This activity demonstrates how to initialize our SDK and integrate it with IronSource.
 */
class IronSourceActivity : Activity() {
    private val customUserId: String? = "..your user id"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_ironsource)

        val builder: SdkBuilder = JustTrackSdkBuilder(this, MainActivity.token)
        if (customUserId != null) {
            // or, if you want to provide your own user id:
            builder.setEnableIronSourceIntegration(true, customUserId)
        } else {
            builder.setEnableIronSourceIntegration(true, IronSourceUserIdSource.JustTrack)
        }
        builder.build()
    }

    /**
     * If you haven't set up the IronSource SDK using "setEnableIronSourceIntegration,"
     * you can integrate IronSource by using "integrateIronSource" after creating the SDK.
     */
    private fun integrateIronSource(sdk: JustTrackSdk) {
        if (customUserId == null) {
            val integrationFuture = sdk.integrateWithIronSource(IronSourceUserIdSource.JustTrack)
            // if you need to know whether the integration succeeded and is done:
            integrationFuture.get()
        } else {
            sdk.integrateWithIronSource(customUserId)
        }
        // if no exception is thrown, the integration was successful
    }

}